### DISCUSSION TITLE
Different session expired error on bulk load method vs normal crud

### DISCUSSION THREAD

**Discussion by Zambonilli** [jsforce/jsforce#1435] [Ideas]
The field and code returned from the bulk API load method, http://jsforce.github.io/jsforce/doc/Bulk.html#load, when a session is expired is different than just calling the various CRUD endpoints.  The bulk api json returned for an invalid session is {"code":"InvalidSessionId"}.  While the json returned from a CRUD endpoint is returned as {"errorCode":"INVALID_SESSION_ID"}.  Ideally these would be the same so client's can use the same logic to handle expired sessions.

**Comment by cristiand391** [comment]
triaging old issues...

this still happens on jsforce v3, the error message comes from the APIs and jsforce just build the error from that. So far only Bulk APIs seem to do this:
https://github.com/jsforce/jsforce/blob/b8eab0f7021d804b24ee4d1a6e1c49729815e313/src/api/bulk.ts#L729

probably worth exploring if we want to start categorizing/normalizing errors in jsforce.
